al-khaser/Anti Debug/NtQueryObject_AllTypesInformation.cpp

```diff
@@ -10,6 +10,8 @@ locate "DebugObject". Todo: Support for Win10
 
 BOOL NtQueryObject_ObjectAllTypesInformation()
 {
+	//NOTE this check is unreliable, a debugger present on the system doesn't mean it's attached to you
+

```

Comment:
Yeah that's a possibility, however it seems useless since we already check
for this with the all objects function. It's your library though, so do
whatever makes more sense to you 😀
On Fri, 1 Jul 2016 at 13:21, Noteworthy <notifications@github.com> wrote:

> In al-khaser/Anti Debug/NtQueryObject_AllTypesInformation.cpp
> <https://github.com/LordNoteworthy/al-khaser/pull/50#discussion_r69285423>
> :
>
> > @@ -10,6 +10,8 @@ locate "DebugObject". Todo: Support for Win10
> >
> >  BOOL NtQueryObject_ObjectAllTypesInformation()
> >  {
> > +	//NOTE this check is unreliable, a debugger present on the system doesn't mean it's attached to you
> > +
>
> I mean by *Same goes for NtQueryObject_ObjectTypeInformation()* that I am
> thankful for comment and fixes. Going back to
> *NtQueryObject_ObjectTypeInformation* method, why can't we say that if
> ObjectInformation->TotalNumberOfObjects == 1, means no debugger and
> everything else, something fishy either a hook or debugger ?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/LordNoteworthy/al-khaser/pull/50/files/7ba77a173e528470f73b6bd25fc9374c04b7dbf4#r69285423>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ACWCmVx7Fpviw3xM-Sj7sRYcFic-vO3xks5qRPg4gaJpZM4I_rAC>
> .
>